### PR TITLE
Migrate social links to use a data file

### PR DESCRIPTION
**Pull request by chrisfinazzo**
Per @DirtyF's comment [here](https://github.com/jekyll/minima/issues/327#issuecomment-451765885).

### CODE REVIEW

File: _includes/social.html [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  {%- if site.microdotblog_username -%}<li><a rel="me" href="https://micro.blog/{{ site.microdotblog_username| cgi_escape | escape }}"><svg class="svg-icon grey"><use xlink:href="{{ '/assets/minima-social-icons.svg#microdotblog' | relative_url }}"></use></svg> <span class="username">{{ site.microdotblog_username| escape }}</span></a></li>{%- endif -%}
-  {%- if site.googleplus_username -%}<li><a href="https://plus.google.com/{{ site.googleplus_username| escape }}"><svg class="svg-icon grey"><use xlink:href="{{ '/assets/minima-social-icons.svg#googleplus' | relative_url }}"></use></svg> <span class="username">{{ site.googleplus_username| escape }}</span></a></li>{%- endif -%}
-  {%- if site.rss -%}<li><a href="{{ 'feed.xml' | relative_url }}"><svg class="svg-icon grey"><use xlink:href="{{ '/assets/minima-social-icons.svg#rss' | relative_url }}"></use></svg> <span>{{ site.rss | escape }}</span></a></li>{%- endif -%}
+{% for social in site.data.social %}
```
Comment (ashmaroli) [inline _includes/social.html]: This looping is wrong. `site.data.social` is not an array of items

**Comment by ashmaroli** [comment]
@chrisfinazzo I would recommend that you model the change here based on #146 as following:
```
{%- assign social = site.data.social -%}
[...]
```
But then since data files inside theme-gems are not read-in by Jekyll, this will require additional changes to be a complete feature.

**Comment by chrisfinazzo** [comment]
Oh, Ok, I see what you're getting at - will update. I knew the loop wasn't right, but at first thought I messed up something w.r.t Mastodon as it is nested inside the others. I have a similar setup on my own site, so that will be something to work off of.

**Comment by DirtyF** [comment]
Maybe we could:

- only provide Github and Twitter by default in config.yml like it was before minima v2.2.0
- if a `social.yaml` file exist, use the items listed in the data file instead
- default social.yaml contains a few social media examples

Document how to add a social service: 
 - add an SVG icon
 - add entry in _data/social.yml

The main idea being minima only offer basic support for social links, and it's up to the user to add more services based on the given examples.

**Comment by ashmaroli** [comment]
@DirtyF If #146 were to merge before this PR, we could simply have
```
{%- assign social = site.data.social | default: site.minima.social_links -%}
```
Either ways, things look like a breaking change..

**Comment by chrisfinazzo** [comment]
@ashmaroli, Yeah, I noticed that we'd be stepping on each other's work here - either way you'd like for me to do this is fine. I'll close the other related PR's when this is done.

**Comment by ashmaroli** [comment]
This needs some documentation now.
It should especially note that Minima will look **either within `site.data.social` or within `site.minima.social_links`**.

As of now, uncommenting an entry in the data file will cause Minima to simply ignore the entire listing in the config file.

**Comment by ashmaroli** [comment]
@chrisfinazzo Please don't pull in other user's contributions into this PR.
When this PR gets merged, the commits will be squashed into one under your name.
The other user's contribution will be lost.

The proper way is to wait till this PR is merged and then help the other user resolve their conflicts or recommend starting a new PR with latest `master` branch.

**Comment by chrisfinazzo** [comment]
> @chrisfinazzo Please don't pull in other user's contributions into this PR.
> When this PR gets merged, the commits will be squashed into one under your name.
> The other user's contribution will be lost.
> 
> The proper way is to wait till this PR is merged and then help the other user resolve their conflicts or recommend starting a new PR with latest `master` branch.

Yeah, this was my fault, I cherry picked that commit out of @baleato's repo and then updated it to match the format, which had changed since then. Merge didn't go right so I just grabbed the files and redid it, which isn't fair to @baleato as it doesn't really credit him properly.

There are a few similar outstanding PR's (#322, #325, #327), I'll ask them to get the latest master and update to the new format before we bring them in.

**Comment by ashmaroli** [comment]
> I'll ask them to get the latest master and update to the new format before we bring them in.

That reminds me of another thing.
Minima was conceived to be a very minimal theme to get the get going with Jekyll and themes.
I'm not sure if adding every social media icon out there boards well with the "minimalistic" principle here.

Instead of adding icon-after-icon, we should simply convert our template into an extensible entity without hampering maintainability.

**Comment by chrisfinazzo** [comment]
I still have the `_data` directory in my local copy and plan to actually do the migration when these few others close. IIRC (please, @DirtyF, correct me if I've got it backwards), doing it this way means people can turn on as many or as few of the services as they want. Not sure how I feel about asking people to write this themselves, even if it isn't all that difficult.

**Comment by ashmaroli** [comment]
My vision is thus:
- convert the social-icon listing into an array of key-value pairs.
- support only a few popular social media
- ask the end-user to procure SVG and the domain format for the remaining social media
- Minima renders whatever icons they list in their config file / data file, if available in their `assets` directory

**Comment by baleato** [comment]
Hi @chrisfinazzo, I've updated #324. I appreciate the effort. I'd suggest not to worry about the other change. I'll keep updating the PR in case of conflicts. That way I'll also get more insight on the latest changes ;)

**Comment by chrisfinazzo** [comment]
> * ask the end-user to procure SVG and the domain format for the remaining social media

This is the key issue I think in determining whether or not the feature gets used. Prior [work][] on this shows that even if users are proficient enough to acquire an SVG for the service they want to use, it doesn't always come out right. This is usually because files are either unavailable in that format, or the XML used to create them is doing something which causes a display issue. 

[work]: https://github.com/jekyll/minima/issues/327#issuecomment-451765885

To that end, I would put this question to @jekyll/minima and @jekyll/stability to determine what the right way forward is.

For my part, I agree with your and @DirtyF's idea in principle, but wonder how much of it should be abstraced via `_data` or asking people to go the DIY route.

**Comment by chrisfinazzo** [comment]
Closing this.

**Pull request closed without merging**